cmd/rb-main.go

```diff
@@ -167,8 +168,13 @@ func deleteBucket(ctx context.Context, url string) *probe.Error {
 		}
 	}()
 
+	foundErr := false
 	for perr := range errorCh {
 		errorIf(perr.Trace(url), "Failed to remove `"+url+"`, continuing..")
+		foundErr = true
+	}
+	if foundErr {
+		return probe.NewError(errors.New("Bucket not removed"))
```

Comment:
Can we return the error returned by errorCh instead of a generic error ?

Reply:
Hi @vadmeste ,

The errors are displayed in the above for loop as they are hit one by one. So, there could be a bunch of error messages on display. 

When we are out of the loop, we may choose to say nothing, show a generic message, or maybe use one of them or the last one but that would repeat an already displayed error message. 
I don' think it would look good to show the last or any one of the displayed error message one more time.

So, I chose to show a generic message, however, we can choose to stop without showing any error message as an alternative.
Please let me know if you think we should go with no error message.

Reply:
@ebozduman, I think we need to quit when the first error is encountered. If we fail to remove something in a bucket, we know in advance we won't be able to remove the bucket, I am not sure why we have the continuing process here.. are we retrying something ? if we do I think we don't have to, minio-go will retry for us.

Basically, I think the best is to have this:
`mc: <Error> Failed to remove mm1/locked: unable to remove xxxx: object is WORM protected and cannot be..`

Reply:
@vadmeste ,

We are not retrying anything. I think this is by design.
I believe the original requirement was to have a single complete pass and show the user all of the errors in one shot, instead of running the same command several times and collecting information about each and every error one by one in single runs.

We can definitely stop at the first error, but that would change the api's behavior. The current implementation cleans up all the prefixes and objects if there is no problem removing them. Stopping at the first error would stop the full cleanup of the non-problematic objects/files.
@kannappanr?

Reply:
@vadmeste,

Actually, the error messages are returned by `deleteBucket` function, so I replaced the code and simplified it such that it returns on the first error as you've requested.

I also changed 2 error messages to better explain the situation. 

When user tries to forcefully remove a bucket, `mc rb --force <alias>/<bucket>`, with one or more `WORM protected` objects, the old error message was not giving the name of the offending object, hence it was too generic. 

Old error message example:
```
mc: <ERROR> Failed to remove `myminio/locked/`. Object is WORM protected
and cannot be overwritten.
```

Now, the name and the version id of the object are part of the error message.
New error message example:
```
mc: <ERROR> Failed to remove `myminio/locked/`. Object, 'prefix/obj1
(Version ID=b1d0b084-9b70-46fe-ae63-a2efdcd3866c)' is WORM protected
and cannot be overwritten.
```